components/ButtonBar/src/Theming/MDCButtonBar+Theming.m

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCButtonBar+Theming.h"
+
+#import "MaterialButtonBar+ColorThemer.h"
+#import "MaterialButtonBar+TypographyThemer.h"
+
+@implementation MDCButtonBar (MaterialTheming)
+
+- (void)applyPrimaryThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme {
+  id<MDCColorScheming> colorScheme = scheme.colorScheme;
+  if (!colorScheme) {
+    colorScheme =
+        [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+  }
+  [self applyPrimaryThemeWithColorScheme:colorScheme];
+
+  id<MDCTypographyScheming> typographyScheme = scheme.typographyScheme;
+  if (!typographyScheme) {
+    typographyScheme =
+        [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
+  }
+  [self applyPrimaryThemeWithTypographyScheme:typographyScheme];
+}
+
+- (void)applyPrimaryThemeWithColorScheme:(nonnull id<MDCColorScheming>)scheme {
+  [MDCButtonBarColorThemer applySemanticColorScheme:scheme toButtonBar:self];
```

Comment:
Is it expected that we need to move the logic from the themer to here when graduating this theming extension to ready? because we want to make sure the central logic is kept in the themer before the migration happens?

Reply:
That's my assumption yeah. For now I think it's better to minimize logic duplication. When we delete the Themer APIs we can move their logic into these methods here.

Reply:
Makes sense:) We just need to make sure to remove all pod dependencies when doing the deprecation.